### PR TITLE
Add mount_option and partition rules

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:

- add following rules:

    - partition_for_boot

    - mount_option_boot_noexec

    - mount_option_boot_noauto

    - partition_for_opt

    - mount_option_opt_nosuid

    - mount_option_srv_nosuid

    - mount_option_home_noexec

    - partition_for_usr

    - mount_option_var_noexec

    

- updated mount_partition_for_srv

#### Rationale:

anssi effort

### CODE REVIEW

File: linux_os/guide/system/software/disk_partitioning/partition_for_usr/rule.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+platform: machine
+
+identifiers:
+    cce@rhel7: CCE-83342-6
+    cce@rhel8: CCE-83343-4
+
+references:
+    anssi: BP28(R12)
+
+{{{ complete_ocil_entry_separate_partition(part="//usr") }}}
```
Comment (yuumasato) [inline linux_os/guide/system/software/disk_partitioning/partition_for_usr/rule.yml:27]: Typo, double stash in `/usr`.

File: linux_os/guide/system/software/disk_partitioning/partition_for_usr/rule.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+title: 'Ensure //usr Located On Separate Partition'
+
+description: |-
+    It is recommended that the <tt>/usr</tt> directory resides on a separate
+    partition.
+
+rationale: |-
+    The <tt>/usr</tt> partition contains system software, utilities and files.
+    PUtting it on a separate partition allows limit its size and apply
```
Comment (yuumasato) [inline linux_os/guide/system/software/disk_partitioning/partition_for_usr/rule.yml:13]: Typo, `Putting` with two upper case letters.

File: linux_os/guide/system/software/disk_partitioning/partition_for_usr/rule.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+documentation_complete: true
+
+prodtype: fedora,ol7,ol8,rhel6,rhel7,rhel8
+
+title: 'Ensure //usr Located On Separate Partition'
```
Comment (yuumasato) [inline linux_os/guide/system/software/disk_partitioning/partition_for_usr/rule.yml:5]: Typo, double slash in `/usr`.

**Review by yuumasato — CHANGES_REQUESTED** [review]
Looks good to me, just a few typos.

**Comment by vojtapolasek** [comment]
Thank you, fixed.

**Comment by jan-cerny** [comment]
@vojtapolasek Please remove RHEL 6 from prodtypes because RHEL 6 was completely removed.

**Review by yuumasato — APPROVED** [review]

**Pull request merged**